plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergQueryRunner.java

```diff
@@ -69,31 +71,90 @@ public static DistributedQueryRunner createIcebergQueryRunner(
             Optional<File> metastoreDirectory)
             throws Exception
     {
-        Session session = testSessionBuilder()
-                .setCatalog(ICEBERG_CATALOG)
-                .setSchema("tpch")
-                .build();
-
-        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session)
+        Builder builder = builder()
                 .setExtraProperties(extraProperties)
-                .build();
+                .setIcebergProperties(connectorProperties)
+                .setInitialTables(tables);
+
+        metastoreDirectory.ifPresent(builder::setMetastoreDirectory);
+        return builder.build();
+    }
 
-        queryRunner.installPlugin(new TpchPlugin());
-        queryRunner.createCatalog("tpch", "tpch");
+    public static Builder builder()
+    {
+        return new Builder();
+    }
 
-        Path dataDir = metastoreDirectory.map(File::toPath).orElseGet(() -> queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data"));
+    public static class Builder
+            extends DistributedQueryRunner.Builder<Builder>
+    {
+        private Optional<File> metastoreDirectory = Optional.empty();
+        private ImmutableMap.Builder<String, String> icebergProperties = ImmutableMap.builder();
```

Comment:
The fact that this is a Builder is a little weird to me, but feel free to ignore.

Reply:
I copied from HiveQueryRunner.Builder. I guess it is ok as we never delete from the map.